core/utils/minhash_utils.hpp

```diff
@@ -27,11 +27,21 @@
 #include <span>
 #include <vector>
 
+#include "utils/math_utils.hpp"
+
 namespace iresearch {
 
 // Implementation of MinHash variant with a single hash function.
 class MinHash {
  public:
+  // Returns number of hashes required to preserve probabilistic error
+  // threshold.
+  static constexpr size_t MaxSize(double_t err) noexcept {
+    return err > 0. && err <= 1.
+               ? math::ceil64(1. / (err * err))
+               : (0. == err ? std::numeric_limits<size_t>::max() : 0);
```

Comment:
```suggestion
    return 0. < err && err <= 1.
               ? math::ceil64(1. / (err * err))
               : (err < 1 ? std::numeric_limits<size_t>::max() : 0);
```

Also why maxsize == 1 if err == 1.0? But if err == 1.0001 then maxsize == 1